### PR TITLE
Make ElemwiseCategorical default.

### PR DESCRIPTION
**Pull request by twiecki**
Fixes https://github.com/pymc-devs/pymc3/issues/1563

### CODE REVIEW

**Comment by springcoil** [comment]
Some test errors with isinstance. Is this an inheritance issue?

File: pymc3/step_methods/metropolis.py [hunk 1]
```diff
@@ -317,6 +317,7 @@ def astep(self, q0, logp):
         q = np.copy(q0)
         logp_curr = logp(q)
 
+        # TODO Replace loop with theano.scan
```
Comment (fonnesbeck) [inline pymc3/step_methods/metropolis.py:320]: Can we do this when `q` is a numpy array?

Comment (twiecki) [reply pymc3/step_methods/metropolis.py:320]: Yeah, that shouldn't be a problem.

**Comment by junpenglao** [comment]
@twiecki do you want to rebase or work on this after 3.1rc4 #2223?

**Comment by twiecki** [comment]
I started having doubts on this. The elemwise sampler is much faster, but for more data points it should never accept a proposal. So I thin it's better to be accurate than fast. There are ways to speed this up. Ultimately though, people should just use the Mixture distributions.

**Comment by junpenglao** [comment]
Sure - the MCMC might need some refactoring anyway in the near future. do you want to also delete the branch?

**Pull request closed without merging**